src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -186,7 +191,7 @@ private BoundExpression TransformPropertyOrEventReceiver(Symbol propertyOrEvent,
             // SPEC VIOLATION: in a case of unconstrained generic type parameter a runtime test (default(T) == null) would be needed
             // SPEC VIOLATION: However, for compatibility with Dev12 we will continue treating all generic type parameters, constrained or not,
             // SPEC VIOLATION: as value types.
-            var variableRepresentsLocation = rewrittenReceiver.Type.IsValueType || rewrittenReceiver.Type.Kind == SymbolKind.TypeParameter;
+            var variableRepresentsLocation = rewrittenReceiver.Type!.IsValueType || rewrittenReceiver.Type.Kind == SymbolKind.TypeParameter;
```

Comment:
>rewrittenReceiver.Type! [](start = 45, length = 23)

How do we know?

Reply:
Generally, it would be an error for a receiver not to have a type.  That is enforced in the binder.  I use that fact in many places to `!` the type of a receiver.  But I will change that to an assertion here and elsewhere.

---
In reply to: [386624089](https://github.com/dotnet/roslyn/pull/42068#discussion_r386624089) [](ancestors = 386624089)